src/cephadm/cephadm

```diff
@@ -7133,34 +7159,36 @@ def check_ssh_connectivity(ctx: CephadmContext) -> None:
         ssh_priv_key_path = f'/tmp/ssh_key_{uuid.uuid1()}'
         ssh_pub_key_path = f'{ssh_priv_key_path}.pub'
         ssh_key_gen_cmd = ['ssh-keygen', '-q', '-t', 'rsa', '-N', '', '-C', '', '-f', ssh_priv_key_path]
-        if subprocess.call(ssh_key_gen_cmd) != 0:
+        _, _, code = call(ctx, ssh_key_gen_cmd)
+        if code != 0:
             logger.warning('Cannot generate keys to check ssh connectivity.')
             return
 
     with open(ssh_pub_key_path, 'r') as f:
         key = f.read()
-    authorize_ssh_key(key, ctx.ssh_user)
+    new_key = authorize_ssh_key(key, ctx.ssh_user)
     ssh_cfg_file_arg = ['-F', pathify(ctx.ssh_config.name)] if ctx.ssh_config else []
-    out, err, code = call(ctx, ['ssh', '-o StrictHostKeyChecking=no',
-                                *ssh_cfg_file_arg, '-i', ssh_priv_key_path,
-                                '-o PasswordAuthentication=no',
-                                f'{ctx.ssh_user}@{get_hostname()}',
-                                'sudo echo'])
-
-    # done, let' remove tmp files and delete pub key from the authorized_files
-    revoke_ssh_key(key, ctx.ssh_user)
-    if ctx.ssh_private_key is None and ctx.ssh_public_key is None:
-        os.remove(ssh_priv_key_path)
-        os.remove(ssh_pub_key_path)
-
+    _, _, code = call(ctx, ['ssh', '-o StrictHostKeyChecking=no',
+                            *ssh_cfg_file_arg, '-i', ssh_priv_key_path,
+                            '-o PasswordAuthentication=no',
+                            f'{ctx.ssh_user}@{get_hostname()}',
+                            'sudo echo'])
+
+    # we only remove the key if it's a new one. In case the user has provided
+    # some already existing key then we don't alter authorized_keys file
+    if new_key:
+        revoke_ssh_key(key, ctx.ssh_user)
+
+    pub_key_msg = '- The public key file configured by --ssh-public-key is valid\n' if ctx.ssh_public_key else ''
+    prv_key_msg = '- The private key file configured by --ssh-private-key is valid\n' if ctx.ssh_private_key else ''
```

Comment:
NIT, but given you're adding a check that the pub/priv ket are either both supplied or neither now, do we only need to check if one of them is set to include both messages?